From e5d56a1cea7684d2c96c94126f5af3dc458c6d73 Mon Sep 17 00:00:00 2001
From: Anantha Krishnan <ananthk@codeaurora.org>
Date: Mon, 15 Apr 2013 10:53:50 +0530
Subject: [PATCH 14/56] bluetooth: Configure the GPIO lines for PCM
 functionality

4 GPIO lines are used for AUX PCM functionality and the lines
have to be configured during BT Turn On sequence and put in
high-Z state during BT Turn Off process.
Signed-off-by: Anantha Krishnan <ananthk@codeaurora.org>

Change-Id: I8405b598ee80bbd54375c5746abc934676d4bce0
Signed-off-by: Anantha Krishnan <ananthk@codeaurora.org>
Signed-off-by: Suman Kumar <suman@inforcecomputing.com>
---
 arch/arm/mach-msm/board-8064-bt.c |   14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-msm/board-8064-bt.c b/arch/arm/mach-msm/board-8064-bt.c
index e4a7fb5..ab9c63d 100644
--- a/arch/arm/mach-msm/board-8064-bt.c
+++ b/arch/arm/mach-msm/board-8064-bt.c
@@ -115,7 +115,7 @@ static int config_pcm(int mode)
 	int pin, rc = 0;
 
 	if (mode == BT_PCM_ON) {
-		pr_err("%s mode =BT_PCM_ON", __func__);
+		pr_debug("%s mode : BT_PCM_ON", __func__);
 		for (pin = 0; pin < ARRAY_SIZE(bt_config_pcm_on);
 			pin++) {
 				rc = gpio_tlmm_config(bt_config_pcm_on[pin],
@@ -124,7 +124,7 @@ static int config_pcm(int mode)
 					return rc;
 		}
 	} else if (mode == BT_PCM_OFF) {
-		pr_err("%s mode =BT_PCM_OFF", __func__);
+		pr_debug("%s mode : BT_PCM_OFF", __func__);
 		for (pin = 0; pin < ARRAY_SIZE(bt_config_pcm_off);
 			pin++) {
 				rc = gpio_tlmm_config(bt_config_pcm_off[pin],
@@ -575,12 +575,22 @@ static int atheros_bluetooth_power(int on)
 			gpio_bt_sys_reset_en);
 		gpio_direction_output(gpio_bt_sys_reset_en, 1);
 		msleep(100);
+		/*setup BT PCM lines*/
+		pr_debug("%s: Configuring PCM lines.\n", __func__);
+		rc = config_pcm(BT_PCM_ON);
+		if (rc < 0)
+			pr_err("%s: config_pcm , rc = %d\n", __func__, rc);
 		goto out;
 	} else {
 		pr_debug("%s: Powering down BT module on AR3002\n", __func__);
 		gpio_set_value(gpio_bt_sys_reset_en, 0);
 		rc = gpio_direction_input(gpio_bt_sys_reset_en);
 		msleep(100);
+		pr_debug("%s: Releasing the PCM lines.\n", __func__);
+		rc = config_pcm(BT_PCM_OFF);
+		if (rc < 0)
+			pr_err("%s: Disabling PCM lines failed with rc =%d\n",
+				__func__, rc);
 	}
 free_gpio:
 	gpio_free(gpio_bt_sys_reset_en);
-- 
1.7.9.5

